bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java

```diff
@@ -114,11 +145,20 @@
      */
     @NotNull
     public Optional<Link<Page>> getLink(@Nullable Page page) {
-        if (page == null) {
+        Page resolved = resolveRedirects(page);
+        if (resolved == null && page != null) {
+            String redirectTarget = page.getProperties().get(PageImpl.PN_REDIRECT_TARGET, "");
```

Comment:
what about the situation where:
page A -> page B -> external URL
looks like this only handles external redirects if they're at the first level, which is the more common scenario.

Reply:
I'll prepare a test case and handling for this as well

Reply:
Should be solved by 56f5eab